src/daemon/ca/certauth.rs

```diff
@@ -932,41 +978,32 @@ impl CertAuth {
     /// Adds a parent. This method will return an error in case a parent
     /// by this name (handle) is already known. Or in case the same reponse
     /// is used for more than one parent.
-    fn add_parent(&self, parent: Handle, info: ParentCaContact) -> KrillResult<Vec<Evt>> {
+    fn add_parent(&self, parent: Handle, info: ParentCaContact) -> KrillResult<Vec<CaEvt>> {
         if self.parent_known(&parent) {
             Err(Error::CaParentDuplicateName(self.handle.clone(), parent))
         } else if let Some(other) = self.parent_for_info(&info) {
             Err(Error::CaParentDuplicateInfo(self.handle.clone(), other.clone()))
         } else if self.is_ta() {
             Err(Error::TaNotAllowed)
         } else {
-            Ok(vec![EvtDet::parent_added(&self.handle, self.version, parent, info)])
+            Ok(vec![CaEvtDet::parent_added(&self.handle, self.version, parent, info)])
         }
     }
 
     /// Removes a parent. Returns an error if it doesn't exist.
-    fn remove_parent(&self, parent: Handle) -> KrillResult<Vec<Evt>> {
-        let _parent = self.parent(&parent)?;
-        let repo = self.repository_contact()?;
-
-        // remove the parent, the RCs and un-publish everything.
-        let mut deltas = vec![];
-        for rc in self.resources.values().filter(|rc| rc.parent_handle() == &parent) {
-            deltas.push(rc.withdraw(repo.repo_info()));
-        }
-
-        Ok(vec![EvtDet::parent_removed(&self.handle, self.version, parent, deltas)])
+    fn remove_parent(&self, parent: Handle) -> KrillResult<Vec<CaEvt>> {
+        Ok(vec![CaEvtDet::parent_removed(&self.handle, self.version, parent)])
```

Comment:
Is the same functionality retained after this change, i.e. it looks like it was unpublishing "everything" but no longer does? Is that intentional? Or is this in some way about recording less events but no changing actual functionality?